### PR TITLE
Do not call info-set guestinfo.ip if ip is empty

### PR DESCRIPTION
**Pull request by caglar10ur**

### CODE REVIEW

**Comment by dougm** [comment]
lgtm

File: pkg/vsphere/toolbox/service.go [hunk 1]
```diff
@@ -223,6 +223,10 @@ func (s *Service) SetOption(args []byte) ([]byte, error) {
 	case "broadcastIP": // TODO: const-ify
 		if val == "1" {
 			ip := s.PrimaryIP()
+			if ip == "" {
```
Comment (hickeng) [inline pkg/vsphere/toolbox/service.go]: what is the value we get back in `netInterfaceAddrs()` in the DefaultIP path - do we need a corresponding code change there as well?

Comment (caglar10ur) [inline pkg/vsphere/toolbox/service.go]: I don't think so, we just end up interface with no ip

File: pkg/vsphere/toolbox/service_test.go [hunk 2]
```diff
@@ -185,7 +185,7 @@ func TestServiceErrors(t *testing.T) {
 
 	in.rpc = []*testRPC{
 		{"Capabilities_Register", "OK "},
-		{"Set_Option broadcastIP 1", "ERR "},
+		{"Set_Option broadcastIP 1", "OK "},
```
Comment (dougm) [inline pkg/vsphere/toolbox/service_test.go]: We want to keep this to cover that error path.  I think to cover the new empty ip check/return path, would need add something different that sets service.PrimaryIP to a func that returns empty.

**Comment by fdawg4l** [comment]
I don't really understand this code but it looks fine.

LGTM

File: pkg/vsphere/toolbox/service.go [hunk 3]
```diff
@@ -223,6 +223,10 @@ func (s *Service) SetOption(args []byte) ([]byte, error) {
...
+				log.Printf("failed to find primary IP\n")
```
Comment (hmahmood) [inline pkg/vsphere/toolbox/service.go]: You should probably make this a warning.

Comment (caglar10ur) [inline pkg/vsphere/toolbox/service.go]: That's is golang's log package so we don't have levels there. Not sure whether @dougm prefers it over logrus.

Comment (dougm) [inline pkg/vsphere/toolbox/service.go]: Right, vsphere/toolbox package currently uses stdlib log.. if we have a good reason to switch it to logsrus, let's do that in a separate PR that changes the rest of the logging too.

**Comment by mdubya66** [comment]
Open for 1 month, is it still valid?

**Comment by caglar10ur** [comment]
Happened again so here is the rebased version with test changes
